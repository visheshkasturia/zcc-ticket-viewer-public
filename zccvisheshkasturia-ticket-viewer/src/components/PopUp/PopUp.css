.popup-box {
  position: fixed;
  background: #00000050;
  width: 100%;
  height: 100vh;
  top: 0;
  left: 0;
}
 
.box {
  position: relative;
  width: 520px;
  margin: 0 auto;
  height: auto;
  margin-top: calc(100vh - 85vh - 20px);
  background: #fff;
  border-radius: 4px;
  padding: 20px;
  border: 1px solid #999;
  overflow: auto;
  display: flex;
  justify-content: center;
  flex-direction: column;
}
 
.close-btn {
  border-radius: 8px;
  height: 40px;
  width: 100px;
  margin-top: 10px;
  margin-left: auto;
  margin-right: auto;
}

.close-btn:hover {
  cursor: pointer;
}