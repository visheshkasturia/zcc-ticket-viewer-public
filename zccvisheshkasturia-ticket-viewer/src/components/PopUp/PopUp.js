import React from 'react';
import './PopUp.css';

function PopUp(props) {
  const prop = props;
  return (
    <div>
      <div className="popup-box">
        <div className="box">
          {prop.content}
          <button data-testid="pop-close-test" type="button" className="close-btn" onClick={prop.handleClose}>Close</button>
        </div>
      </div>
    </div>
  );
}

export default PopUp;
