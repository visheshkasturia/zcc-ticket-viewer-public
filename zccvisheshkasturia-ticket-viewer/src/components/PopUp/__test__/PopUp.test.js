import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PopUp from '../PopUp';

describe('PopUp renders correctly', () => {
  it('Button has correct text', () => {
    const { getByTestId } = render(<PopUp />);
    expect(getByTestId('pop-close-test')).toHaveTextContent('Close');
  });

  it('Matches snapshot', () => {
    const tree = renderer.create(<PopUp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
