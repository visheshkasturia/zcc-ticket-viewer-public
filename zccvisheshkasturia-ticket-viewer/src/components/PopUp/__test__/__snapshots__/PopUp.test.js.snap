// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp renders correctly Matches snapshot 1`] = `
<div>
  <div
    className="popup-box"
  >
    <div
      className="box"
    >
      <button
        className="close-btn"
        data-testid="pop-close-test"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
